ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1631,6 +1633,9 @@ public Table apply(org.apache.hadoop.hive.metastore.api.Table table) {
       }
     } catch (Exception e) {
       throw new HiveException(e);
+    } finally {
+      long diff = System.nanoTime() - t1;
+      LOG.debug(String.format(logString, "getTablesByType", diff, dbName, ""));
```

Comment:
Can we also make the logging conditional based on `(LOG.isDebugEnabled())`

Reply:
Fwiw the perf logger already has a check on the logging level?

Reply:
I will still add `(LOG.isDebugEnabled())` just to be extra careful